lib/_http_outgoing.js

```diff
@@ -125,6 +127,19 @@ function OutgoingMessage() {
 }
 util.inherits(OutgoingMessage, Stream);
 
+Object.defineProperty(OutgoingMessage.prototype, 'socket', {
+  get: function() {
+    return this[kSocket];
+  },
+  set: function(socket) {
+    this[kSocket] = socket;
+    if (socket && socket[async_id_symbol] === undefined) {
+      socket[async_id_symbol] = async_hooks.newUid();
```

Comment:
I think that is a completely different problem, this is built to be a stop-gap fixes that avoid crashing node 8 everywhere during unit tests.
The line below sets the type to `'not a socket'`, which is probably not correct in that case.